toolkit/tools/roast/formats/squashfs.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package formats
+
+import (
+	"fmt"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+// SquashFSType represents the squashfs format
+const SquashFSType = "squashfs"
+
+// SquashFS implements Converter interface to convert a RAW image into a squashfs file
+type SquashFS struct {
+}
+
+// Convert converts the image in the squashfs format
+func (t *SquashFS) Convert(input, output string, isInputFile bool) (err error) {
+	const (
+		squashErrors = false
+	)
+
+	if !isInputFile {
+		return fmt.Errorf("squashfs conversion requires an input")
+	}
+
+	err = shell.ExecuteLive(squashErrors, "mksquashfs", input, output, "-all-root", "-wildcards", "-e", "tmp/*", "-noappend")
```

Comment:
For general image usage in the toolkit are `-all-root` and `-e tmp/*` appropriate? The former seems to clobber ownership of files/dirs in the image being converted, which may not be the right call for all images. As for excluding `tmp/*`, I'm wondering if the image format converter may not be the best place in the flow to specify that it should be removed.

Reply:
totally agree. I've removed those options.